harness/determined/common/api/_session.py

```diff
@@ -99,3 +99,14 @@ def put(
         timeout: Optional[int] = None,
     ) -> requests.Response:
         return self._do_request("PUT", path, params, json, data, headers, timeout, False)
+
+    def with_retry(self, retry: Optional[urllib3.util.retry.Retry]) -> "Session":
```

Comment:
question: why an optional retry w/o any defaults? what would be the use case of providing a None retry? It could be useful if we provided a safe default 🤔

Reply:
This is for explicitly overriding the retry behavior, so if somebody calls `with_retry(None)` we should absolutely respect that.